### DISCUSSION TITLE
Disabled options and/or option groups with labels?

### DISCUSSION THREAD

**Discussion by excid3** [josefarias/hotwire_combobox#126] [Ideas]
Was using Hotwire Combobox today for Hatchbox.io to help users set up cron expressions and realized it would be really handy to have disabled options much like a select tag supports. We were primarily trying to use that as a label like an option group in this case, but disabling options in selects is common too.

Thinking it could be as simple as a `disabled` attribute on the item? Maybe you would need `pointer-events: none` if disabled to prevent interaction?

**Comment by josefarias** [comment]
Ah yes this makes sense. Not sure what the API would look like. Like, if you pass an `ActiveRecord::Relation` as the options, how would the combobox know which ones are disabled?

I'm sure it's a solvable problem though, just articulating it.

I'd welcome a PR attempting this (for anyone reading this) — feel free to respond here with ideas about how we could implement. If not, I might get to it eventually.

**Reply by excid3** [reply]
`options_for_select` has this example, but I believe the last argument in the array can also add HTML attributes because I've added data attributes to them before.

```ruby
options_for_select(["Free", "Basic", "Advanced", "Super Platinum"], disabled: ["Advanced", "Super Platinum"])
```

```ruby
options_for_select(["Free", "free", {disabled: true}])
```

`grouped_options_for_select` is similar too, but takes a Hash.
